cmake/nsis/FileAssociation.nsh

```diff
@@ -128,6 +128,7 @@ NoBackup:
   StrCmp $0 "" 0 Skip
     WriteRegStr HKCR "$R0" "" "$R0"
     WriteRegStr HKCR "$R0\shell" "" "open"
+    WriteRegStr HKCR "$R0\DefaultIcon" "" "$R2,1"
```

Comment:
You can move this one line down if you want to always create this key.

Reply:
So, if I put it under Skip, it will always create it?

Reply:
Yes.

Reply:
> So, if I put it under Skip, it will always create it?

I believe `SKIP:` will only create it on reinstall, never on new install.   I wouldn't recommend changing this logic, personally.  We need to fix reinstall behavior and this is only a small portion of it.  We can just write these settings by hand every time.  That's how I do it in other projects.

Reply:
According to the [NSIS documentation](http://nsis.sourceforge.net/Reference/StrCmp), The behavior is:
- If `$0` is empty(i.e. key `$R0` doesn't exist), continue executing
- If not(key `$R0` already exists), jump to `Skip`

Reply:
Thanks.  So, what part about my evaluation is incorrect?

Reply:
You can interpret the `StrCmp` line as `if (strcmp($0,"") != 0) goto Skip;`. So lines under the label will always be executed IIUC.